### PR TITLE
SI-7618 Remove octal number literals

### PR DESCRIPTION
**Pull request by soc**
This also fixes a spurious detection of octal literals in floating
point literals: Literals like 01.23 are neither deprecated nor planned
for removal.

### CODE REVIEW

File: src/compiler/scala/tools/nsc/ast/parser/Scanners.scala [hunk 1]
```diff
@@ -981,9 +972,11 @@ trait Scanners extends ScannersCommon {
     protected def getNumber() {
       val base1 = if (base < 10) 10 else base
       // read 8,9's even if format is octal, produce a malformed number error afterwards.
+      var notSingleZero = false
       while (digit2int(ch, base1) >= 0) {
         putChar(ch)
         nextChar()
+        notSingleZero = true
```
Comment (adriaanm) [inline src/compiler/scala/tools/nsc/ast/parser/Scanners.scala:982]: Could you add a comment on why this flag can soundly be computed this way? The logic is quite spread out, or it's at least not obvious to me.

Comment (soc) [inline src/compiler/scala/tools/nsc/ast/parser/Scanners.scala:982]: Will do.

**Comment by soc** [comment]
Review by @adriaanm, please! I added comments. Hopefully things are a bit more clear now.

**Comment by adriaanm** [comment]
LGTM -- thanks!

**Pull request merged**